src/cli/settings/unset.rs

```diff
@@ -20,26 +20,40 @@ pub struct SettingsUnset {
 
 impl SettingsUnset {
     pub fn run(self) -> Result<()> {
-        let path = if self.local {
-            config::local_toml_config_path()
-        } else {
-            config::global_config_path()
-        };
-        let raw = file::read_to_string(&path)?;
-        let mut config: DocumentMut = raw.parse()?;
-        if !config.contains_key("settings") {
-            return Ok(());
-        }
-        let settings = config["settings"].as_table_mut().unwrap();
-        settings.remove(&self.key);
-
-        // validate
-        let _: SettingsFile = toml::from_str(&config.to_string())?;
-
-        file::write(&path, config.to_string())
+        unset(&self.key, self.local)
     }
 }
 
+pub fn unset(mut key: &str, local: bool) -> Result<()> {
+    let path = if local {
+        config::local_toml_config_path()
+    } else {
+        config::global_config_path()
+    };
+    let raw = file::read_to_string(&path)?;
+    let mut config: DocumentMut = raw.parse()?;
+    if !config.contains_key("settings") {
+        return Ok(());
+    }
+    let mut settings = config["settings"].as_table_mut().unwrap();
```

Comment:
```suggestion
    if let Some(mut settings) = config["settings"].as_table_mut() {
```

that way we can remove the unwrap